tests/test_client_functional.py

```diff
@@ -2927,6 +2927,73 @@ async def test_creds_in_auth_and_url() -> None:
         await session.close()
 
 
+@pytest.mark.parametrize(
+    ["check_redirect_auth_precedence"],
+    ([False], [True]),
+    ids=("redirect auth clash allowed", "redirect auth precedence"),
+)
+async def test_creds_in_auth_and_redirect_url(
+    create_server_for_url_and_handler: Callable[[URL, Handler], Awaitable[TestServer]],
+    check_redirect_auth_precedence: bool,
+) -> None:
+    url_from = URL("http://example.com")
+    url_to = URL("http://user@example.com")
+    redirected = False
+
+    async def srv(request: web.Request) -> web.Response:
+        nonlocal redirected
+
+        assert request.host == url_from.host
+
+        if not redirected:
+            redirected = True
```

Comment:
@webknjaz Changed this as well when updating the test so the precedence test is able to fail separately if the precedence fix is undone, otherwise both tests just end up raising TooManyRequests due to the artificial infinite redirect, which is suboptimal.